src/server/schemas/wallet/index.ts

```diff
@@ -20,8 +21,25 @@ export const walletWithAAHeaderSchema = Type.Object({
     ...AddressSchema,
     description: "Smart account address",
   }),
+  "x-account-factory-address": Type.Optional({
+    ...AddressSchema,
+    description: "Smart account factory address",
+  }),
 });
 
+/*
+ * Helper function to parse an address string.
+ * Returns undefined if the address is invalid.
+ */
+export function maybeAddress(address: string | undefined) {
+  if (!address) return undefined;
+  try {
+    return getAddress(address);
```

Comment:
Any reason to silently ignore invalid addresses? I think if an invalid address is passed, the expectation should be to error instead of assume no address. That way the developer can fix it instead of wondering why the tx looks like it's going through but not working.

Reply:
we already have `getAddress` that errors out. I built the interface such that if an incorrect address is passed into `x-account-factory-address`, it will ignore the header. This felt necessary for v1 for backwards compatibility reasons. Now we can make it fail loudly too. Changing it 👍